src/cogent3/app/data_store_new.py

```diff
@@ -356,6 +356,22 @@ def __init__(
         self._source_check_create(mode)
         self._limit = limit
 
+    def __getitem__(self, index):
+        if isinstance(index, Union[int, slice]):
+            return self.members[index]
+        elif isinstance(index, str):
+            for member in self.completed:
+                if member.unique_id == index:
+                    return member
+            for member in self.not_completed:
+                if member.unique_id == f"{_NOT_COMPLETED_TABLE}/{index}.json":
+                    return member
+            raise IndexError("Datamember index is not defined")
```

Comment:
@GavinHuttley 
I am not sure about this line. Should we return None or raise an error?